package/MDAnalysis/coordinates/base.py

```diff
@@ -1082,54 +1162,70 @@ def _sliced_iter(self, start, stop, step):
         """
         # override with an appropriate implementation e.g. using self[i] might
         # be much slower than skipping steps in a next() loop
-        def _iter(start=start, stop=stop, step=step):
-            try:
-                for i in xrange(start, stop, step):
-                    yield self._read_frame(i)
-            except TypeError:  # if _read_frame not implemented
-                raise TypeError("{0} does not support slicing."
-                                "".format(self.__class__.__name__))
-        return _iter()
-
-    def _check_slice_indices(self, start, stop, step):
-        """Unpack the slice object and do checks
+        try:
+            for i in xrange(start, stop, step):
+                yield self._read_frame(i)
+        except TypeError:  # if _read_frame not implemented
+            raise TypeError("{0} does not support slicing."
+                            "".format(self.__class__.__name__))
+
+    def check_slice_indices(self, start, stop, step):
+        """Check frame indices are valid and clip to fit trajectory
+
+        Parameters
+        ----------
+        start, stop, step
+          Values representing the slice indices.
+          Can use `None` to use defaults of (0, -1, and 1)
+          respectively.
 
-        Return the start stop and step indices
+        Returns
+        -------
+        start, stop step
+          Integers representing the slice
         """
-        if not (((type(start) == int) or (start is None)) and
-                ((type(stop) == int) or (stop is None)) and
-                ((type(step) == int) or (step is None))):
-            raise TypeError("Slice indices are not integers")
+        for var, varname in (
+                (start, 'start'),
+                (stop, 'stop'),
+                (step, 'step')
+        ):
```

Comment:
I personally don't like that the loop starting `:` is at the same indentation as the `for` keywords. I would prefer a formating like this.

```python
 for var, varname in ((start, 'start'),
                      (stop, 'stop'),
                      (step, 'step')):
```